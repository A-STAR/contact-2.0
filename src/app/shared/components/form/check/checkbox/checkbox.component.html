<input type="checkbox" (change)="onChange()"/>
