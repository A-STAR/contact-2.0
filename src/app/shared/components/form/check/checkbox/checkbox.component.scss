@import '../../../../styles/app/variables.scss';

label {
  line-height: 18px;
}

span {
  background-color: #fff;
  border: 1px solid $gray;
  border-radius: 2px;
  cursor: pointer;
  display: inline-block;
  height: 18px;
  margin: 0;
  margin-right: 4px;
  position: relative;
  transition: background-color 0.2s ease, border-color 0.2s ease;
  vertical-align: bottom;
  width: 18px;

  &:hover {
    border-color: $link-color;
  }

  &.checked {
    background-color: $link-color;
    border-color: $link-color;

    &:before,
    &:after {
      background-color: #fff;
      bottom: 4px;
      content: ' ';
      left: 7px;
      position: absolute;
      transform-origin: bottom left;
      width: 1px;
    }

    &:before {
      height: 6px;
      transform: rotate(-45deg);
    }

    &:after {
      height: 10px;
      transform: rotate(35deg);
    }
  }

  &.disabled {
    background-color: $gray-light;
    border-color: $gray;
    cursor: default;
    &:before,
    &:after {
      background-color: $gray-dark;
    }
  }
}

input {
  left: -10000px;
  position: absolute;
  visibility: hidden;
}
