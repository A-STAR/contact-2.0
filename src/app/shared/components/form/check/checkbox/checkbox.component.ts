import {
  ChangeDetectionStrategy,
  ChangeDetectorRef,
  Component,
  forwardRef,
  HostBinding,
  HostListener,
} from '@angular/core';
import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers: [
    {
      provide: NG_VALUE_ACCESSOR,
      useExisting: forwardRef(() => CheckboxComponent),
      multi: true,
    }
  ],
  selector: 'app-checkbox',
  styleUrls: [ './checkbox.component.scss' ],
  template: ''
})
export class CheckboxComponent {
  private _value: boolean;

  constructor(
    private cdRef: ChangeDetectorRef,
  ) {}

  @HostBinding('class.checked')
  get value(): boolean {
    return this._value;
  }

  writeValue(value: boolean): void {
    this._value = null;
    this.cdRef.markForCheck();
  }

  registerOnChange(fn: Function): void {
    this.propagateChange = fn;
  }

  registerOnTouched(fn: Function): void {
    // No need in touch callback for checkbox
    // because a click will change its value and mark control as dirty anyway
  }

  @HostListener('click')
  onClick(): void {
    this._value = !this._value;
    this.propagateChange(this._value);
    this.cdRef.markForCheck();
  }

  private propagateChange: Function = () => {};
}

