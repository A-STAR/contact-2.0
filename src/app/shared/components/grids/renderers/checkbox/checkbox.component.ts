import { ChangeDetectionStrategy, Component } from '@angular/core';
import { ICellRendererParams } from 'ag-grid/main';
import { ICellRendererAngularComp } from 'ag-grid-angular';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  selector: 'app-grid-checkbox-renderer',
  template: `<app-checkbox [ngModel]="value" (ngModelChange)="onChange($event)"></app-checkbox>`,
})
export class CheckboxRendererComponent implements ICellRendererAngularComp {
  private params: ICellRendererParams;

  agInit(params: ICellRendererParams): void {
    this.params = params;
  }

  refresh(): boolean {
    return false;
  }

  get value(): boolean {
    return this.params.value;
  }

  onChange(value: boolean): void {
    this.params.node.setDataValue(this.params.column, value);
  }
}
