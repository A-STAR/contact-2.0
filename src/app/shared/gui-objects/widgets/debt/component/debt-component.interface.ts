export interface IDebtComponent {
  id: number;
  typeCode: number;
  sum: number;
  currencyId: number;
}
