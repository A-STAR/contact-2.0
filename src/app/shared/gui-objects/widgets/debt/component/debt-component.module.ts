import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DebtComponentGridModule } from './grid/debt-component-grid.module';

import { DebtComponentService } from './debt-component.service';

@NgModule({
  imports: [
    DebtComponentGridModule,
    CommonModule,
  ],
  exports: [
    DebtComponentGridModule,
  ],
  providers: [
    DebtComponentService,
  ]
})
export class DebtComponentModule { }
