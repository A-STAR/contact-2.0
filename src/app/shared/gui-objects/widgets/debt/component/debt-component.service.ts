import { Injectable } from '@angular/core';
import { Observable } from 'rxjs/Observable';

import { IDebtComponent } from './debt-component.interface';

import { DataService } from '../../../../../core/data/data.service';
import { NotificationsService } from '../../../../../core/notifications/notifications.service';

@Injectable()
export class DebtComponentService {
  static MESSAGE_DEBT_COMPONENT_SAVED = 'MESSAGE_DEBT_COMPONENT_SAVED';

  constructor(
    private dataService: DataService,
    private notificationsService: NotificationsService,
  ) {}

  fetchAll(debtId: number): Observable<Array<IDebtComponent>> {
    return this.dataService
      .read('/debts/{debtId}/components', { debtId })
      .map(response => response.components)
      .catch(this.notificationsService.error('errors.default.read').entity('entities.debtComponents.gen.plural').dispatchCallback());
  }

  fetch(debtId: number, debtComponentId: number): Observable<IDebtComponent> {
    return this.dataService
      .read('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId })
      .map(response => response.components[0])
      .catch(this.notificationsService.error('errors.default.read').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  create(debtId: number, debtComponent: IDebtComponent): Observable<void> {
    return this.dataService
      .create('/debts/{debtId}/components', { debtId }, debtComponent)
      .catch(this.notificationsService.error('errors.default.create').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  update(debtId: number, debtComponentId: number, debtComponent: Partial<IDebtComponent>): Observable<void> {
    return this.dataService
      .update('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId }, debtComponent)
      .catch(this.notificationsService.error('errors.default.update').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }

  delete(debtId: number, debtComponentId: number): Observable<void> {
    return this.dataService
      .delete('/debts/{debtId}/components/{debtComponentId}', { debtId, debtComponentId })
      .catch(this.notificationsService.error('errors.default.delete').entity('entities.debtComponents.gen.singular').dispatchCallback());
  }
}
