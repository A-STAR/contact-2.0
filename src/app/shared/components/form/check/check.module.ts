import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';

import { CheckboxComponent } from './checkbox/checkbox.component';
import { RadioButtonComponent } from './radiobutton/radiobutton.component';

@NgModule({
  imports: [
    CommonModule,
    FormsModule,
  ],
  exports: [
    CheckboxComponent,
    RadioButtonComponent,
  ],
  declarations: [
    CheckboxComponent,
    RadioButtonComponent,
  ],
})
export class CheckModule {}
