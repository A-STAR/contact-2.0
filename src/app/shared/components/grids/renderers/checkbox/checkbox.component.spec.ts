import { async as Async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';

import { CheckModule } from '@app/shared/components/form/check/check.module';

import { CheckboxCellRendererComponent } from './checkbox.component';

describe('CheckboxCellRendererComponent', () => {
  let fixture: ComponentFixture<CheckboxCellRendererComponent>;

  beforeEach(Async(() => {
    TestBed
      .configureTestingModule({
        imports: [
          CheckModule,
          FormsModule,
        ],
        declarations: [
          CheckboxCellRendererComponent,
        ],
      })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CheckboxCellRendererComponent);
  });

  it('should render checkbox renderer', () => {
    fixture.componentInstance.agInit({
      node: {
        setDataValue: () => {},
      },
      value: false,
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
