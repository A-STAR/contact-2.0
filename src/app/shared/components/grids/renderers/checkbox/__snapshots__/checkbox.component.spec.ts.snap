// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxCellRendererComponent should render checkbox renderer 1`] = `
<div
  id="root0"
  ng-version="5.2.4"
>
  
      
  <label>
    
        
    <app-checkbox
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-model="false"
    >
      <input
        type="checkbox"
      />
      

    </app-checkbox>
    
      
  </label>
  
    
</div>
`;
